tests/testsuite/check.rs

```diff
@@ -1005,6 +1005,37 @@ fn warn_manifest_with_project() {
         .run();
 }
 
+#[cargo_test(nightly, reason = "edition2024")]
```

Comment:
Nightly gate is for rustc. I don't think we need the gate here.

```suggestion
```

Reply:
Ignore this sugggestion if you want to keep edition tests for consistency reason.

Reply:
Honestly, I worry about that.

I had assumed we pass the new edition down but we only pass the lint group.  Its a bit odd that we have the lint group working on stable.